src/query/tsdb/remote/compressed_codecs.go

```diff
@@ -113,7 +113,9 @@ func compressedTagsFromTagIterator(tagIter ident.TagIterator, encoderPool serial
 		return nil, fmt.Errorf("no refs available to data")
 	}
 
-	return data.Bytes(), nil
+	db := data.Bytes()
+	// Need to copy the encoded bytes to a buffer as the encoder keeps a reference to them
```

Comment:
wanna use pooled bytes rather than alloc here?

TBH: this is one of the reasons we end up with so many wrapper objects in the m3db codebase - because we need a place to track what pool an object should be returned to once we're done with it (can't use just a slice). 

Will need to use [GRPC interceptors](https://github.com/grpc/grpc-go/blob/master/interceptor.go#L77) to release back to the pool safely

Reply:
This was more of an afterthought after seeing issues caused by this on the internal side; this is just fixing it as is, rather than the correct approach... Added [an issue](https://github.com/m3db/m3/issues/1015) to track this properly since it's out of scope of this PR and probably a decent chunk of work (as soon as we start intercepting GRPC calls, might as well add pools for all the proto slices we're building, and would prefer to keep that to a dedicated PR)

Reply:
fair enough, mind adding a `// TODO(arnikola): pool this when implementing https://github.com/m3db/m3/issues/1015` as a reminder.

Reply:
No worries 👍